graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.tsx

```diff
@@ -186,9 +186,35 @@ const IndexSetConfigurationForm = ({
   submitLoadingText,
 } : Props) => {
   const history = useHistory();
-  const [indexSetState] = useState<IndexSet>(indexSet);
+  const [indexSet, setIndexSet] = useState(initialIndexSet);
+
   const [fieldTypeRefreshIntervalUnit, setFieldTypeRefreshIntervalUnit] = useState<Unit>('seconds');
   const { loadingIndexDefaultsConfig, indexDefaultsConfig } = useIndexDefaults();
+
+  useEffect(() => {
+    if (loadingIndexDefaultsConfig || !indexDefaultsConfig) return;
+
```

Comment:
I think here it makes sense to create a hook useIndexSet that contains all the useEffect logic + useState, to keep the component light and readable it already has a lot of lines and functions, what do you think?

Reply:
Good idea, did that. Can you test it again?